File: Sum Root to Leaf Numbers.cpp
#define F(i, a, b) for(int i = a; i <= b; i++)
#define FB(i, a, b) for(int i = a; i >= b; i--)
#define MAX2(a, b) a>b?a:b
#define MIN2(a, b) a<b?a:b

#include <iostream>
#include <vector>
#include <queue>
#include <stack>

/**
 * Definition for binary tree
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    void helper( TreeNode* node, int tmp, int& sum ){
        tmp *= 10;
        tmp += node->val;

        if( node->left == NULL && node->right == NULL ){
            sum += tmp;
            return ;
        }

        if( node->left != NULL ){
            helper( node->left, tmp, sum );
        }

        if( node->right != NULL ){
            helper( node->right, tmp, sum );
        }

        return ;
    }

    int sumNumbers(TreeNode *root) {
        int sum = 0;
        if( root == NULL ){
            return sum;
        }

        helper( root, 0, sum );
        return sum;
    }
};

File: LeetCode-Java/Sum Root to Leaf Numbers.java
/**
 * Definition for binary tree
 * public class TreeNode {
 *	 int val;
 *	 TreeNode left;
 *	 TreeNode right;
 *	 TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
	int ret = 0;
	int pathSum = 0;
	public void helper(TreeNode root){
		pathSum *= 10;
		pathSum += root.val;
		if( root.left == null && root.right == null ){
			ret += pathSum; 
		}else{
			if( root.left != null ){
				helper( root.left );
			}
			if( root.right != null ){
				helper( root.right );
			}
		}
		pathSum /= 10; 
	}

	public int sumNumbers(TreeNode root) {
		if( root == null ){
			return 0;
		}
		ret = 0;
		pathSum = 0;

		helper(root);

		return ret; 
	}
}

